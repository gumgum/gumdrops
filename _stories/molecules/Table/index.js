/* eslint-disable react/no-multi-comp */
import React from 'react';
import readme from './README.md';
import { boolean, number } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { optionalSelect } from '../../../components/utils/optionalSelect';
import Table from '../../../components/molecules/Table';
import TableData from '../../../components/molecules/TableData';
import ToolTip from '../../../components/atoms/Tooltip';

const jobTitles = [
    'Motel Maid',
    'Twister Operator',
    'Butane Compressor Operator',
    'Optomechanical Technician',
    'Automobile Body Repair Supervisor',
    'Underground Bolting Machine Operator',
    'Funeral Arranger',
    'Cellist',
    'Labor Economist',
    'Cast-Iron Drain Pipe Layer'
];

const names = [
    'Roselle',
    'Jena',
    'Monnie',
    'Ena',
    'Tonja',
    'Christie',
    'Gretta',
    'Clora',
    'Sona',
    'Magan',
    'Sybil',
    'Elly',
    'Shakira',
    'Carri',
    'Piedad',
    'Valorie',
    'Melinda',
    'Tracie',
    'Brynn',
    'Flossie'
];

const sizeOptions = {
    xs: 'xs',
    sm: 'sm',
    lg: 'lg',
    xl: 'xl',
    'No Value': ''
};

// Column options

const columnsSimple = ['id', 'name', 'title', 'age', 'weight', 'height'];

const columnsCustom = [
    { key: 'id', children: 'ID' },
    { key: 'name', children: 'First Name' },
    { key: 'title', children: 'Job Title' },
    { key: 'age', children: 'Age (years)' },
    { key: 'weight', children: 'Weight (pounds)' },
    { key: 'height', children: 'Height' }
];

const weightCellDecorator = (cellData, key) => (
    <TableData
        key={key}
        context={cellData < 200 ? 'positive' : cellData === 200 ? 'neutral' : 'negative'}>
        {cellData < 200 ? `↓ ${cellData}` : cellData === 200 ? cellData : `↑ ${cellData}`}
    </TableData>
);

const heightCellDecorator = (cellData, key) => (
    <TableData key={key} context={!cellData ? 'negative' : undefined}>
        {cellData ? cellData : 'n/a'}
    </TableData>
);

const columnsAdvanced = [
    { key: 'id', children: 'ID' },
    {
        key: 'name',
        children: (
            <ToolTip text="Uses a custom click handler">
                First Name <i className="fa fa-user -m-r-2" />
            </ToolTip>
        ),
        disableSorting: true,
        headingProps: {
            onClick: action('column_heading_clicked')
        }
    },
    {
        key: 'title',
        children: (
            <ToolTip text="Uses a custom click handler">
                Job Title (sorted by length) <i className="fa fa-info-circle -m-r-2" />
            </ToolTip>
        ),
        headingProps: {
            onClick: action('column_heading_clicked')
        },
        sortCompareAsc: (a, b) => a.title.length - b.title.length,
        sortCompareDesc: (a, b) => b.title.length - a.title.length
    },
    { key: 'age', children: 'Age (years)' },
    {
        key: 'weight',
        children: 'Weight (pounds)',
        dataCellDecorator: weightCellDecorator
    },
    {
        key: 'height',
        children: 'Height',
        headingProps: {
            style: {
                width: '120px'
            },
            className: 'custom-class'
        },
        dataCellDecorator: heightCellDecorator
    }
];

// Column select options

const columnOptions = {
    simple: 'simple',
    custom: 'custom',
    advanced: 'advanced'
};

const columnConfig = {
    simple: columnsSimple,
    custom: columnsCustom,
    advanced: columnsAdvanced
};

// Sample data

function getRandomInt(min, max) {
    return Math.floor(Math.random() * (max - min) + min);
}

const createPerson = () => ({
    name: names[getRandomInt(0, names.length - 1)],
    title: jobTitles[getRandomInt(0, jobTitles.length - 1)],
    age: getRandomInt(18, 70),
    weight: getRandomInt(110, 250),
    height: Math.random() < 0.9 ? `${getRandomInt(4, 7)}'${getRandomInt(1, 11)}"` : null
});

const generateData = amount =>
    Array(amount)
        .fill(null)
        .map((_, i) => ({
            id: i,
            ...createPerson()
        }));

const component = () => {
    const numOfRows = number('Rows', 10, {
        range: true,
        min: 1,
        max: 100,
        step: 1
    });

    return (
        <Table
            columns={columnConfig[optionalSelect('Column Config', columnOptions, 'simple')]}
            data={generateData(numOfRows)}
            hasHeader={boolean('hasHeader', true)}
            isInverse={boolean('isInverse', false)}
            isSecondary={boolean('isSecondary', false)}
            isStriped={boolean('isStriped', false)}
            isResponsive={boolean('isResponsive', false)}
            onRowClick={boolean('onRowClick', false) ? action('row_clicked') : undefined}
            size={optionalSelect('Size', sizeOptions, '')}
        />
    );
};

export default [readme, component];
