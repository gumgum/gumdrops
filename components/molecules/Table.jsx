import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

// table components
import Header from './TableHeader';
import Heading from './TableHeading';
import Body from './TableBody';
import Row from './TableRow';
import Data from './TableData';
import { deprecateLog } from '../utils/deprecate';

const sortDirection = {
    ASC: 'asc',
    DESC: 'desc'
};

class Table extends Component {
    constructor(props) {
        super(props);
        this.state = {
            data: props.data,
            sortBy: {}
        };
    }

    componentDidUpdate(prevProps) {
        if (prevProps.data !== this.props.data) {
            /* eslint-disable react/no-did-update-set-state */
            this.setState({
                data: this.props.data
            });
        }
    }

    _sortCompareAlpha = (a, b, key) => {
        if (!a[key]) return -1;
        if (!b[key]) return 1;
        const textA = a[key].toUpperCase();
        const textB = b[key].toUpperCase();
        return textA < textB ? -1 : textA > textB ? 1 : 0;
    };

    _sortDescending({ key, sortCompareDesc }) {
        const data = [...this.props.data];
        const shouldSortByAlpha = typeof data[0][key] === 'string'; // assumes data consistency across array items

        const sortedData = sortCompareDesc
            ? data.sort(sortCompareDesc)
            : shouldSortByAlpha
              ? data.sort((a, b) => this._sortCompareAlpha(b, a, key))
              : data.sort((a, b) => b[key] - a[key]);

        this.setState({
            data: sortedData,
            sortBy: { key, direction: sortDirection.DESC }
        });
    }

    _sortAscending({ key, sortCompareAsc }) {
        const data = [...this.props.data];
        const shouldSortByAlpha = typeof data[0][key] === 'string'; // assumes data consistency across array items

        const sortedData = sortCompareAsc
            ? data.sort(sortCompareAsc)
            : shouldSortByAlpha
              ? data.sort((a, b) => this._sortCompareAlpha(a, b, key))
              : data.sort((a, b) => a[key] - b[key]);

        this.setState({
            data: sortedData,
            sortBy: { key, direction: sortDirection.ASC }
        });
    }

    _sortColumn = config => {
        const { sortBy: { direction, key: stateKey } } = this.state;
        const { key } = config;

        const shouldSortDesc = direction === sortDirection.ASC && key === stateKey;

        return shouldSortDesc ? this._sortDescending(config) : this._sortAscending(config);
    };

    renderTableContent({
        columns,
        isStriped,
        isInverse,
        hasHeader,
        onRowClick,
        isSecondary,
        customRowKey
    }) {
        const { sortBy, data } = this.state;
        return (
            <Fragment>
                {hasHeader && (
                    <Header>
                        <Row>
                            {columns.map((column, i) => {
                                const {
                                    sortCompareAsc,
                                    sortCompareDesc,
                                    disableSorting,
                                    onHeadingClick,
                                    headingProps = {}
                                } = column;

                                const reactKey = `row-${i}-key-${i}`;
                                const key = column.key || column;
                                const children = column.children || column;

                                // NOTE: onHeadingClick should be removed in the next major version
                                // Using "headingProps" will allow more complete customization
                                if (onHeadingClick) {
                                    deprecateLog(
                                        '`onHeadingClick` has been deprecated in favor of `headingProps`. Use `headingProps.onClick` instead. See https://github.com/gumgum/gumdrops/blob/master/_stories/molecules/Table/README.md#columns'
                                    );
                                }

                                const onClickFn = onHeadingClick || headingProps.onClick;

                                const onClickHeading = disableSorting
                                    ? onClickFn
                                    : event => {
                                          this._sortColumn({
                                              key,
                                              sortCompareAsc,
                                              sortCompareDesc
                                          });
                                          onClickFn && onClickFn(event);
                                      };

                                return (
                                    <Heading
                                        isSecondary={isSecondary}
                                        key={reactKey}
                                        onClick={onClickHeading}
                                        sortDirection={
                                            sortBy.key === key ? sortBy.direction : undefined
                                        }
                                        {...headingProps}>
                                        {children}
                                    </Heading>
                                );
                            })}
                        </Row>
                    </Header>
                )}
                <Body>
                    {data.map((rowData, i) => {
                        const rowKey =
                            customRowKey && rowData[customRowKey] ? rowData[customRowKey] : i;
                        return (
                            <Row
                                key={`row-${rowKey}`}
                                className={onRowClick && '-cursor--pointer'}
                                isInverse={isInverse}
                                isStriped={isStriped}
                                onClick={onRowClick && (() => onRowClick(rowData))}>
                                {columns.map((column, k) => {
                                    const isString = typeof column === 'string';
                                    const cellData = isString
                                        ? rowData[column]
                                        : rowData[column.key];
                                    const decorator = column.dataCellDecorator;
                                    const key = `row-${rowKey}-key-${k}`;
                                    return decorator ? (
                                        decorator(cellData, key, rowData)
                                    ) : (
                                        <Data key={key}>{cellData}</Data>
                                    );
                                })}
                            </Row>
                        );
                    })}
                </Body>
            </Fragment>
        );
    }

    renderTable() {
        const { data, className, children, size, isStriped, isInverse } = this.props;
        return (
            <table
                className={cx('gds-table', className, {
                    [`gds-table--${size}`]: size,
                    ['gds-table--inverse']: isInverse,
                    ['gds-table--striped']: isStriped && !isInverse,
                    ['gds-table--inverse-striped']: isStriped && isInverse
                })}>
                {data ? this.renderTableContent(this.props) : children}
            </table>
        );
    }

    render() {
        const { isResponsive } = this.props;
        return isResponsive ? (
            <div className="gds-table--responsive">{this.renderTable()}</div>
        ) : (
            this.renderTable()
        );
    }
}

Table.propTypes = {
    // necessary for controlled tables
    data: function(props, propName, componentName) {
        if (props.data) {
            if (!Array.isArray(props.data)) {
                return new Error(
                    'Invalid prop `' +
                        propName +
                        '` supplied to' +
                        ' `' +
                        componentName   +
                        '`, expected `array`.'
                );
            }
            for (let i = 0; i < props.data.length; i++) {
                if (typeof props.data[i] !== 'object') {
                    return new Error(
                        'Invalid prop `data[' +
                            i +
                            ']` supplied to' +
                            ' `' +
                            componentName +
                            '`, expected `object`.'
                    );
                } else if (props.customRowKey && !props.data[i][props.customRowKey]) {
                    return new Error(
                        'Invalid prop `data[' +
                            i +
                            ']` supplied to' +
                            ' `' +
                            componentName +
                            '`, expected an `object` with property `' +
                            props.customRowKey +
                            '`.'
                    );
                }
            }
        }
    },
    columns: PropTypes.arrayOf(
        PropTypes.oneOfType([
            // columns can be a simple array of strings representing the keys
            // or an object for customizations like unique titles, custom
            // ascending or descending sorts.
            PropTypes.string,
            // customizations
            PropTypes.shape({
                children: PropTypes.node,
                dataCellDecorator: PropTypes.func,
                disableSorting: PropTypes.bool,
                key: PropTypes.string,
                sortCompareAsc: PropTypes.func,
                sortCompareDesc: PropTypes.func,
                // NOTE: deprecated
                onHeadingClick: PropTypes.func,
                headingProps: PropTypes.object
                // callback: PropTypes.func,
                // A callback could be used to request sorted data from the server,
                // for instance paginated items sorted by data.
            })
        ])
    ),

    children: PropTypes.node,
    className: PropTypes.string,
    hasHeader: PropTypes.bool,
    isInverse: PropTypes.bool,
    isSecondary: PropTypes.bool,
    isStriped: PropTypes.bool,
    isResponsive: PropTypes.bool,
    onRowClick: PropTypes.func,
    customRowKey: PropTypes.string,
    size: PropTypes.oneOf(['sm', 'lg', 'xs', 'xl'])
};

Table.defaultProps = {
    hasHeader: true
};

export default Table;
