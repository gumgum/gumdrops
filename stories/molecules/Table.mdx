import { Canvas, Source } from '@storybook/blocks';
import { ArgsTable } from '@storybook/addon-docs';
import * as TableStories from './Table.stories';
import Table from '../../components/molecules/Table';
import TableData from '../../components/molecules/TableData';
import TableHeading from '../../components/molecules/TableHeading';

# Table

**Note:** The table component requires `React ^16.3.0`

The `<Table>` component creates a table from a provided data source. It allows for column configuration, alphanumeric sorting, custom sorting, and theming.

## Simple Usage

Simple usage of the `<Table>` just requires `data` and `columns` props. Data must be provided as an array of objects with keys that match column identifiers. Columns are also provided by an array that identifies the keys of the data to use. Array order will determine their order in the table. Here's an example:

```js
const data = [
    {
        name: 'John',
        age: 36
    },
    {
        name: 'Jane',
        age: 32
    }
];

const columns = ['name', 'age']; // table columns will be ordered by the array

const People = () => <Table columns={columns} data={data} />;
```

## Advanced Column Examples

The `columns` prop config can take a number of more advanced configurations as seen above. For more advanced configuration you can supply additional properties to the `columns` array:

| property               | type       | description                                                                                                                                             |
| ---------------------- | ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------------- |
| key                    | `String`   | Key representing the corresponding data property.                                                                                                       |
| children               | `Node`     | Custom markup to render in the column heading [See example](#custom-column-markup-example)                                                                      |
| disableSorting         | `Boolean`  | Disables ability to sort column data                                                                                                                    |
| dataCellDecorator      | `Function` | Customize cell markup with a decorator, a function that is called with the cell data, a React key, and the row data. [See example](#custom-cell-markup-example) |
| headingProps           | `Object`   | Props object to be passed to the `<TableHeading>` component. [See example](#custom-heading-props-example)                                                       |
| sortCompareAsc         | `Function` | Custom ascending sort compare function. [See example](#custom-sorting-example)                                                                                  |
| sortCompareDesc        | `Function` | Custom descending sort compare function. [See example](#custom-sorting-example)                                                                                 |
| renderExpandableColumn | `Function` | Convert the cell to contain a button that will expand the returned content returned from this method. [See example](#custom-expandable-column)            |
| renderExpandableTrigger | `Function` | Customize the cell used to trigger expansion. Called with the cell data, key, row data, column object, and an object containing `isExpanded` and `onToggle`. The returned element will automatically receive the `chevron-expander` class. |

### Custom Column Markup Example

If custom markup is needed in the column heading, use the `children` property:

```jsx
const columns = [
    {
        key: 'name',
        children: 'Employee Name'
    },
    {
        key: 'age',
        children: <ToolTip text="Custom markup">Age (years)</ToolTip>
    }
];

const People = () => <Table columns={columns} data={data} />;
```

### Custom Cell Markup Example

Custom markup can be created for individual table data cells (`<td>`) by adding a `dataCellDecorator` function:

```jsx
import TableData from 'gumdrops/TableData';

const weightCellDecorator = (cellData, key, rowData) => (
    <TableData
        key={key}
        context={cellData < 200 ? 'positive' : cellData === 200 ? 'neutral' : 'negative'}>
        <a href={`/people/${rowData.name}`}>
            {cellData < 200 ? `↓ ${cellData}` : cellData === 200 ? cellData : `↑ ${cellData}`}
        </a>
    </TableData>
);

const columns = [
    { key: 'name', children: 'First Name' },
    { key: 'title', children: 'Job Title' },
    { key: 'age', children: 'Age (years)' },
    {
        key: 'weight',
        children: 'Weight (pounds)',
        dataCellDecorator: weightCellDecorator
    },
    { key: 'height', children: 'Height' }
];

const People = () => <Table columns={columns} data={data} />;
```

### Custom Sorting Example

Sorting can be customized with the `sortCompareAsc` and `sortCompareDesc` properties which define the compare functions to be used for ascending and descending sorting.

```jsx
const columnsAdvanced = [
    { key: 'name' },
    {
        key: 'title',
        children: 'Job Title (sorted by title length)',
        sortCompareAsc: (a, b) => a.title.length - b.title.length,
        sortCompareDesc: (a, b) => b.title.length - a.title.length
    },
    { key: 'age' },
    { key: 'weight' },
    { key: 'height' }
];

const People = () => <Table columns={columns} data={data} />;
```

### Custom Heading Props Example

Props can be passed to the `<TableHeading>` component with the `headingProps` property. This is useful when you need to size columns or sort the table data externally such as from an API.

```jsx
const columnsAdvanced = [
    { key: 'name' },
    {
        key: 'title',
        disableSorting: true,
        headingProps: {
            className: 'custom-class',
            style: {
                width: '10%'
            },
            onClick: this.sortColumn('title'), // sorting handled externally
            sortDirection: this.state.sortDirection // one of 'asc' or 'desc'
        }
    },
    { key: 'age' },
    { key: 'weight' },
    { key: 'height' }
];

const People = () => <Table columns={columns} data={data} />;
```

### Custom Row Key Example

Default behaviour is to generate each row's key using a sequential iterator, (row-0, row-1, etc).
If the Table has a `customRowKey` prop, this value will be generated from the column with the same name.

The following example will produce rows with keys `key-x123` and `key-y456` from data rows `id: x123` and `id: y456` respectively.

```jsx
const data = [
    { name: 'Alice', age: 20, id: 'x123' },
    { name: 'Bob', age: 20, id: 'y456' }
];

const People = () => <Table columns={columns} data={data} customRowKey="id" />;
```

### Custom Expandable Column

Each column can take a renderExpandableColumn function which will render a row underneath the column and will allow the user to show
an expandable section below when toggled.

Note: You can have more than one renderExpandableColumn per row and they're handled independently.
Note: If you have sorting enabled, you'll need to provide a `customRowKey` to the table to allow the expanded section to link to the row correctly.

The following example will produce a table with 2 rows and every row's first column will contain a "chevron" that the user
can click to expand the content.

```jsx
const data = [
    { name: 'Alice', age: 20, id: 'x123' },
    { name: 'Bob', age: 20, id: 'y456' }
];
const columns = [{
    key: 'name',
    children: 'Name',
    renderExpandableColumn(cellData, key, rowData, column) {
        return <p>Hi my name is {cellData} and I'm {rowData.age} years old.</p>
    }
    ,
    renderExpandableTrigger(cellData, key, rowData, column, { isExpanded, onToggle }) {
        return (
            <TableData key={key} onClick={onToggle}>
                {cellData} <i className={`fa fa-chevron-${isExpanded ? 'up' : 'down'}`} />
            </TableData>
        );
    }
}, {
    key: 'age',
    children: 'Age',
}]

const People = () => <Table columns={columns} data={data} customRowKey="id" />;
```
Note: custom triggers will automatically receive the `chevron-expander` class so they match the default expand animation.

### Table Footer

The table footer can be used to display totals or other summary information. The footer is rendered as a `<tfoot>` element.

The footer will allow you to have a "static" row at the bottom of the table that will not be affected by sorting.

The footer prop can accept JSX, an object, or an array of objects containing key/value pairs where the "keys" are the column keys.

The following example will create a table with a footer that will display the total age of all the people in the table.

```jsx
const columns = [
    { key: 'name', children: 'First Name' },
    { key: 'age', children: 'Age (years)' },
];
const data = [
    { name: 'Alice', age: 20, id: 'x123' },
    { name: 'Bob', age: 24, id: 'y456' }
];
const footer = {
    name: 'Total Age',
    age: data.reduce((acc, row) => acc + row.age, 0) // -> 44
}
const People = () => <Table columns={columns} data={data} footer={footer} customRowKey="id" />;
```

Multiple footer rows can be provided by passing an array of objects:

```jsx
const multiFooter = [
    {
        name: 'Average Age',
        age: data.reduce((acc, row) => acc + row.age, 0) / data.length
    },
    {
        name: 'Total Count',
        age: data.length
    }
];

const People = () => <Table columns={columns} data={data} footer={multiFooter} customRowKey="id" />;
```


### Custom Table Footer

We can also include JSX to render a custom table footer by importing the components manually.

```jsx
import TableFooter from 'gumdrops/TableFooter';
import TableRow from 'gumdrops/TableRow';
import TableData from 'gumdrops/TableData';
const columns = [
    { key: 'name', children: 'First Name' },
    { key: 'age', children: 'Age (years)' },
];
const data = [
    { name: 'Alice', age: 20, id: 'x123' },
    { name: 'Bob', age: 24, id: 'y456' }
];

function CustomFooter() {
    return <TableFooter>
        <TableRow>
            <TableData colSpan={2}>
                The average age is {data.reduce((acc, row) => acc + row.age, 0) / data.length}
            </TableData>
        </TableRow>
    </TableFooter>
}
const People = () => <Table columns={columns} data={data} footer={CustomFooter} customRowKey="id" />;

```


## Component Example Simple

<Canvas of={TableStories.Simple} />

## Component Example Source Simple

<Source of={TableStories.Simple} />

## Component Example Advanced

<Canvas of={TableStories.Advanced} />

## Component Example Source Advanced

<Source of={TableStories.Advanced} />

## Component Example Custom

<Canvas of={TableStories.Custom} />

## Component Example Source Custom

<Source of={TableStories.Custom} />

## Component Example Expandable

<Canvas of={TableStories.Expandable} />

## Component Example Source Expandable

<Source of={TableStories.Expandable} />

## Component Example Footer

<Canvas of={TableStories.Footer} />

## Component Example Source Footer

<Source of={TableStories.Footer} />

## Table Props

<ArgsTable of={Table} />

## TableData Props

<ArgsTable of={TableData} />

## TableHeading Props

<ArgsTable of={TableHeading} />
