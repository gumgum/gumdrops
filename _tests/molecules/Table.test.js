/* globals mount */
import React from 'react';
import Table from '../../components/molecules/Table';

const defaultProps = {
    data: [
        {
            foo: 'Foo',
            bar: 'Bar'
        }
    ],
    columns: ['foo', 'bar'],
    className: 'custom-class',
    hasHeader: false,
    isInverse: false,
    isSecondary: false,
    isStriped: false,
    isResponsive: false,
    onRowClick: () => {},
    size: 'lg'
};

describe('Expect <Table>', () => {
    it('to render', () => {
        const wrapper = mount(<Table {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('to render customized', () => {
        const props = {
            ...defaultProps,
            hasHeader: true,
            isInverse: true,
            isSecondary: true,
            isStriped: true,
            isResponsive: true
        };
        const wrapper = mount(<Table {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('to render using advanced configs', () => {
        const props = {
            ...defaultProps,
            columns: [
                {
                    key: 'foo',
                    children: 'Foo',
                    headingProps: { style: { width: 100, onClick: () => {} } }
                },
                {
                    key: 'bar',
                    children: 'Bar',
                    headingProps: { style: { width: 100, onClick: () => {} } }
                }
            ]
        };
        const wrapper = mount(<Table {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
