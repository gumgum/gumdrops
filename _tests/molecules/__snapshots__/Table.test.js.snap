// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Table> to render 1`] = `
<Table
  className="custom-class"
  columns={
    Array [
      "foo",
      "bar",
    ]
  }
  data={
    Array [
      Object {
        "bar": "Bar",
        "foo": "Foo",
      },
    ]
  }
  hasHeader={false}
  isInverse={false}
  isResponsive={false}
  isSecondary={false}
  isStriped={false}
  onRowClick={[Function]}
  size="lg"
>
  <table
    className="gds-table custom-class gds-table--lg"
  >
    <TableBody>
      <tbody
        className=""
      >
        <TableRow
          className="-cursor--pointer"
          isInverse={false}
          isStriped={false}
          key="row-0"
          onClick={[Function]}
        >
          <tr
            className="gds-table__row -cursor--pointer"
            onClick={[Function]}
          >
            <TableData
              key="row-0-key-0"
            >
              <td
                className=""
              >
                Foo
              </td>
            </TableData>
            <TableData
              key="row-0-key-1"
            >
              <td
                className=""
              >
                Bar
              </td>
            </TableData>
          </tr>
        </TableRow>
      </tbody>
    </TableBody>
  </table>
</Table>
`;

exports[`Expect <Table> to render customized 1`] = `
<Table
  className="custom-class"
  columns={
    Array [
      "foo",
      "bar",
    ]
  }
  data={
    Array [
      Object {
        "bar": "Bar",
        "foo": "Foo",
      },
    ]
  }
  hasHeader={true}
  isInverse={true}
  isResponsive={true}
  isSecondary={true}
  isStriped={true}
  onRowClick={[Function]}
  size="lg"
>
  <div
    className="gds-table--responsive"
  >
    <table
      className="gds-table custom-class gds-table--lg gds-table--inverse gds-table--inverse-striped"
    >
      <TableHeader>
        <thead
          className=""
        >
          <TableRow>
            <tr
              className="gds-table__row"
            >
              <TableHeading
                isSecondary={true}
                key="row-0-key-0"
                onClick={[Function]}
              >
                <th
                  className="gds-table__header -text-left gds-table__header--secondary"
                  onClick={[Function]}
                >
                  foo
                </th>
              </TableHeading>
              <TableHeading
                isSecondary={true}
                key="row-1-key-1"
                onClick={[Function]}
              >
                <th
                  className="gds-table__header -text-left gds-table__header--secondary"
                  onClick={[Function]}
                >
                  bar
                </th>
              </TableHeading>
            </tr>
          </TableRow>
        </thead>
      </TableHeader>
      <TableBody>
        <tbody
          className=""
        >
          <TableRow
            className="-cursor--pointer"
            isInverse={true}
            isStriped={true}
            key="row-0"
            onClick={[Function]}
          >
            <tr
              className="gds-table__row -cursor--pointer gds-table__row--inverse gds-table__row--striped-inverse"
              onClick={[Function]}
            >
              <TableData
                key="row-0-key-0"
              >
                <td
                  className=""
                >
                  Foo
                </td>
              </TableData>
              <TableData
                key="row-0-key-1"
              >
                <td
                  className=""
                >
                  Bar
                </td>
              </TableData>
            </tr>
          </TableRow>
        </tbody>
      </TableBody>
    </table>
  </div>
</Table>
`;

exports[`Expect <Table> to render using advanced configs 1`] = `
<Table
  className="custom-class"
  columns={
    Array [
      Object {
        "children": "Foo",
        "headingProps": Object {
          "style": Object {
            "onClick": [Function],
            "width": 100,
          },
        },
        "key": "foo",
      },
      Object {
        "children": "Bar",
        "headingProps": Object {
          "style": Object {
            "onClick": [Function],
            "width": 100,
          },
        },
        "key": "bar",
      },
    ]
  }
  data={
    Array [
      Object {
        "bar": "Bar",
        "foo": "Foo",
      },
    ]
  }
  hasHeader={false}
  isInverse={false}
  isResponsive={false}
  isSecondary={false}
  isStriped={false}
  onRowClick={[Function]}
  size="lg"
>
  <table
    className="gds-table custom-class gds-table--lg"
  >
    <TableBody>
      <tbody
        className=""
      >
        <TableRow
          className="-cursor--pointer"
          isInverse={false}
          isStriped={false}
          key="row-0"
          onClick={[Function]}
        >
          <tr
            className="gds-table__row -cursor--pointer"
            onClick={[Function]}
          >
            <TableData
              key="row-0-key-0"
            >
              <td
                className=""
              >
                Foo
              </td>
            </TableData>
            <TableData
              key="row-0-key-1"
            >
              <td
                className=""
              >
                Bar
              </td>
            </TableData>
          </tr>
        </TableRow>
      </tbody>
    </TableBody>
  </table>
</Table>
`;
